make/Docs.gmk

```diff
@@ -264,7 +264,7 @@ define create_overview_file
   $$($1_OVERVIEW): $$($1_OVERVIEW_VARDEPS_FILE)
 	$$(call LogInfo, Creating overview.html for $1)
 	$$(call MakeDir, $$(@D))
-	$$(PRINTF) > $$@ '$$($1_OVERVIEW_TEXT)'
+	$$(PRINTF) "%s" '$$($1_OVERVIEW_TEXT)' > $$@
```

Comment:
For my education, why isn't this just `$$(ECHO) '$$($1_OVERVIEW_TEXT)' > $$@` ?

Reply:
That would add an extra \n at the end. Maybe it does not matter so much, but I did not want to change anything.

Reply:
`echo -n` ?

Reply:
I wasn't aware of that feature in `echo`, but I agree that if we are using printf just to omit the line break, then `echo -n` seems like a better choice.

Reply:
I'm not sure I agree. Why would:
```
$$(ECHO) -n '$$($1_OVERVIEW_TEXT)' 
```
be better than
```
$$(PRINTF) "%s" '$$($1_OVERVIEW_TEXT)' 
```
?

We're kind of squabbling over details here, and it really doesn't matter as much, but I don't think there is really anything to gain by going down that route. If nothing else, `echo -n` seems like less widespread knowledge (Erik didn't know, I knew but had forgot since I never use it). In contrast, I believe the printf syntax is a bit more well known.

But sure, I can use `echo -n` instead of `printf` where possible. Just give me one good reason why.

Reply:
I just thought that if you tried to standardized on `echo` where possible instead of `printf`, it would make sense. I often feel awkward having to switch to the completely different syntax of printf just to prevent the newline. Having a simple option to echo that just omits the newline seems more straightforward. There are of course still usecases for printf. I don't mind that much either way though, so I'm really fine with either.